extensions/transferto/action.go

```diff
@@ -56,25 +56,25 @@ func (a *TransferAirtimeAction) AllowedFlowTypes() []flows.FlowType {
 }
 
 // Execute runs this action
-func (a *TransferAirtimeAction) Execute(run flows.FlowRun, step flows.Step) error {
+func (a *TransferAirtimeAction) Execute(run flows.FlowRun, step flows.Step, logModifier func(flows.Modifier), logEvent func(flows.Event)) error {
 	contact := run.Contact()
 	if contact == nil {
-		run.LogEvent(step, events.NewErrorEvent(errors.Errorf("can't execute action in session without a contact")))
+		logEvent(events.NewErrorEvent(errors.Errorf("can't execute action in session without a contact")))
```

Comment:
a bit more functional - the benefit of which is that now actions don't need to know where to log events and modifiers... they just log them to the passed in functions.